main.go

```diff
@@ -36,12 +51,51 @@ type Firehose struct {
 	processSettings ProcessSettings
 	storageBackend  StorageBackend
 	hedwigConsumer  *hedwig.QueueConsumer
+	messageBuffer []*hedwig.Message
+	flushLock sync.Mutex
+	flushCh chan struct{}
+}
+
+func (fp *Firehose) WriteMessage(wg *sync.WaitGroup, message *hedwig.Message) error {
+	defer wg.Done()
+	// TODO: serialization for hedwig.Message to []byte
+	var data []byte
+	// TODO: how to get store appenv? should file name end with time?
+	currentTime := time.Now()
+	uploadLocation := fmt.Sprintf("%s/%s/%s/%s", "store-appenv?", message.Type, currentTime.Format("2006/1/2"), "filetime?")
```

Comment:
what should the filename be and how do we get the store appenv for the file path? I'm currently planning for the path to be `store-appenv/message-type/YYYY/M/D/filetime`

Reply:
as per out [discussion](https://standardcognition.quip.com/as75AzpK2itI#temp:C:INc7e7e6ca5c99444d0a74f7d2ca) `OutputBucket` is the bucket + path. And in standard specific deploy.sh, we can configure output path to include store-appenv

Reply:
oh this is for the staging bucket, my bad. In any case, I think `store-appenv` is standard specific, so how about we just use the subscription id as an a path instead of `store-appenv`?